oneview/resource_volume.go

```diff
@@ -0,0 +1,336 @@
+// (C) Copyright 2016 Hewlett Packard Enterprise Development LP
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+package oneview
+
+import (
+	"github.com/HewlettPackard/oneview-golang/ov"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVolumeCreate,
+		Read:   resourceVolumeRead,
+		Update: resourceVolumeUpdate,
+		Delete: resourceVolumeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"category": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"device_volume_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"device_specific_attributes": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"copy_state": {
+							Type:     schema.TypeString,
+							Computed: true,
```

Comment:
Are these attributes not set while update as well?

Reply:
its read only